### PR TITLE
fix(tier4_planning_rviz_plugin): supress initial warning message

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

Without this PR, the following warning message is shown on the terminal when launching autoware.launch.xml or turning on the visualization of the path marker.
```
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'View Drivable Area')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Alpha')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Color')
[rviz2-38] QObject::connect: No such slot rviz_common::_RosTopicDisplay::updateVisualization()
[rviz2-38] QObject::connect:  (sender name:   'Width')
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@purewater0901 cc @yukkysaito Could you approve the PR.

File: common/tier4_planning_rviz_plugin/include/path/display.hpp [hunk 1]
```diff
@@ -82,19 +82,14 @@ class AutowarePathWithDrivableAreaDisplay : public AutowarePathBaseDisplay<T>
...
-    property_drivable_area_alpha_->setMax(1.0);
-    property_drivable_area_color_ = new rviz_common::properties::ColorProperty(
-      "Color", QColor(0, 148, 205), "", property_drivable_area_view_, SLOT(updateVisualization()),
-      this);
-    property_drivable_area_width_ = new rviz_common::properties::FloatProperty(
-      "Width", 0.3f, "", property_drivable_area_view_, SLOT(updateVisualization()), this);
-    property_drivable_area_width_->setMin(0.001);
+    property_drivable_area_alpha_.setMin(0.0);
+    property_drivable_area_alpha_.setMax(1.0);
+    property_drivable_area_width_.setMin(0.001);
```
Comment (yukkysaito) [inline common/tier4_planning_rviz_plugin/include/path/display.hpp:92]: If `SLOT(updateVisualization()) is removed`, I think visualization is not updated after parameters are changed.

**Review by yukkysaito — APPROVED** [review]
LGTM

File: common/tier4_planning_rviz_plugin/include/path/display.hpp (hunk 1 above)
Comment (takayuki5168) [reply common/tier4_planning_rviz_plugin/include/path/display.hpp:94]: (Let me unresolve the comment not to forget the reason not to use `SLOT(updateVisualization)` :pray: )
I had a discussion with Saito-san internally.

- In order to register the `updateVisualization` function as `SLOT`, the class that has the function has `Q_OBJECT` type.
    - FYI: `Error: Class declaration lacks Q_OBJECT macro.`
- The template class cannot have `Q_OBJECT` type.
   - FYI: `Error: Template classes not supported by Q_OBJECT.`
- The derived class from the template class with the `updateVisualiation` function can be compiled, but the runtime error `No such slot rviz_common::_RosTopicDisplay::updateVisualization()` occurs.

In short, by the change to use a template class in order to make path/trajectory/... implementation common, now we cannot use  `SLOT(updateVisualization)`.
